src/main/java/org/hydev/mcpm/client/database/SuperLocalPluginTracker.java

```diff
@@ -0,0 +1,619 @@
+     * @param pluginDirectory The path to the plugin directory
+     */
+    public SuperLocalPluginTracker(String mainLockFile, String pluginDirectory) {
+        this.mainLockFile = mainLockFile;
+        this.pluginDirectory = pluginDirectory;
+    }
+
+    /**
+     * Read metadata from a plugin's jar
+     *
+     * @param jar Local plugin jar path
+     * @return Metadata
+     */
+    private PluginYml readMeta(File jar) {
+        try (PluginJarFile InstancePluginJarFile = new PluginJarFile(jar)) {
+            return InstancePluginJarFile.readPluginYaml();
+        } catch (Exception e) {
+            System.out.println("Error reading plugin.yml from " + jar);
+            return null;
+        }
+    }
+
+    /**
+     * Read the CSV file and return a mapping between plugin names and install
+     * status
+     *
+     * @return Mapping between plugin name and boolean status
+     */
+    private ArrayList<PluginTrackerModel> readCsv() {
+        ArrayList<PluginTrackerModel> accumulator = new ArrayList<>();
+        try {
+            CSVReader reader = new CSVReader(new FileReader(mainLockFile));
+            String[] line;
+            while ((line = reader.readNext()) != null) {
+                accumulator.add(new PluginTrackerModel(line[0], Boolean.parseBoolean(line[1]), line[2], line[3]));
+            }
+            reader.close();
+            return accumulator;
+        } catch (FileNotFoundException e) {
+            return accumulator;
+        } catch (CsvException | IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Read a JSON file at the given location file and return a mapping between
+     * plugin names and install
+     * status
+     *
+     * @param csv The mapping between plugin name and boolean status
+     */
+    private ArrayList<PluginTrackerModel> readJson() {
+        try {
+            // ObjectMapper mapper = new ObjectMapper();
+            // Reads the json file and converts it to a list of PluginTrackerModel objects
+            return this.mapper.readValue(new File(mainLockFile), mapper.getTypeFactory()
+                    .constructCollectionType(ArrayList.class, PluginTrackerModel.class));
+        } catch (FileNotFoundException e) {
+
+            return (new ArrayList<PluginTrackerModel>());
+        } catch (JsonParseException | JsonMappingException e) {
+            throw new RuntimeException(e);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+    }
+
+    /**
+     * Save an ArrayList's contents into a JSON file by replacing all contents
+     *
+     * @param list ArrayList of PluginTrackerModel instances to save
+     *
+     */
+    private void saveJson(ArrayList<PluginTrackerModel> list) {
+        try {
+            this.mapper.writeValue(Paths.get(mainLockFile).toFile(), list);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        } catch (Exception ex) {
```

Comment:
I believe the only exception here is IOException, so we also shouldn't catch Exception